### PR TITLE
Opensource examples - Runpod w/Text-Generation-WebUI API Endpoint

### PR DESCRIPTION
**Pull request by PhiBrandon**
Adding Runpod example that utlizes a premade Ubuntu LLM Container on Runpod, allowing the ability to quickly create an endpoint for the variety of GPTQ models available via TheBloke[https://huggingface.co/TheBloke], with Openai compatible endpoint.

Example utilzes Openhermes 2.5 Mistral 7B - GPTQ model. 
Removed Maybe BaseModel, response model = UserDetail.

## Summary by CodeRabbit

- **Documentation**
  - Updated instructions for setting up Openrouter API endpoint and Runpod account.
  - Added details on deploying a template with Runpod.
  - Clarified the addition of versioning to the Runpod API endpoint.
  - Specified that the default Runpod API key should be set to "None".

- **New Features**
  - Introduced a new `UserDetail` class to enhance data handling within the application.

### CODE REVIEW

File: examples/open_source_examples/runpod.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    class UserDetail(BaseModel):
+        name: str = Field(description="Name extracted from the text")
+        age: int = Field(description="Age extracted from the text")
+        occupation: Optional[str] = Field(
+            default=None, description="Occupation extracted from the text"
+        )
+
+    for content in data:
+        prompt_template = f"""
```
Comment (jxnl) [inline examples/open_source_examples/runpod.py:51]: This does not make sense to have a prompt template and then use messages

Comment (PhiBrandon) [reply examples/open_source_examples/runpod.py]: Revised it to properly use messages.

**Review by jxnl — APPROVED** [review]
is this good to merge? can you make sure to run `ruff format .` and then let me know when you're happy to merge

**Comment by PhiBrandon** [comment]
> is this good to merge? can you make sure to run `ruff format .` and then let me know when you're happy to merge

it's good to go!

**Pull request merged**